tools/topology/sof/pipe-eq-fir-volume-capture.m4

```diff
@@ -0,0 +1,131 @@
+# Passthrough with volume Pipeline and PCM
+#
+# Pipeline Endpoints for connection are :-
+#
+#  host PCM_C <-- B0 <-- Volume 0 <-- B1 <-- EQ-FIR <-- B2 <-- source DAI0
+
+# Include topology builder
+include(`utils.m4')
+include(`buffer.m4')
+include(`pcm.m4')
+include(`pga.m4')
+include(`dai.m4')
+include(`mixercontrol.m4')
+include(`pipeline.m4')
+include(`bytecontrol.m4')
+include(`eq_fir.m4')
+
+#
+# Controls
+#
+# Volume Mixer control with max value of 32
+C_CONTROLMIXER(Master Capture Volume, PIPELINE_ID,
+	CONTROLMIXER_OPS(volsw,
+		256 binds the mixer control to volume get/put handlers,
+		256, 256),
+	CONTROLMIXER_MAX(, 80),
+	false,
+	CONTROLMIXER_TLV(TLV 80 steps from -50dB to +30dB for 1dB, vtlv_m50s1),
+	Channel register and shift for Front Left/Right,
+	LIST(`	', KCONTROL_CHANNEL(FL, 1, 0), KCONTROL_CHANNEL(FR, 1, 1)))
+
+#
+# Volume Configuration
+#
+
+define(DEF_PGA_TOKENS, concat(`pga_tokens_', PIPELINE_ID))
+define(DEF_PGA_CONF, concat(`pga_conf_', PIPELINE_ID))
+
+W_VENDORTUPLES(DEF_PGA_TOKENS, sof_volume_tokens,
+LIST(`		', `SOF_TKN_VOLUME_RAMP_STEP_TYPE	"0"'
+     `		', `SOF_TKN_VOLUME_RAMP_STEP_MS		"250"'))
+
+W_DATA(DEF_PGA_CONF, DEF_PGA_TOKENS)
+
+#
+# FIR EQ
+#
+
+define(DEF_EQFIR_COEF, concat(`eqfir_coef_', PIPELINE_ID))
+define(DEF_EQFIR_PRIV, concat(`eqfir_priv_', PIPELINE_ID))
+
+# Fred: define filter set
```

Comment:
I was adding more comments here, That changes are missed. I will collect all the review comments and process them together.